### PR TITLE
Patch event reduce errors

### PR DESCRIPTION
**Pull request by TristanH**
These will be properly fixed by Daniel as he resolves this issue: https://github.com/pubkey/event-reduce/issues/444

But in the mean time, at least make sure our queries aren't incorrect!

### CODE REVIEW

File: src/event-reduce.ts [hunk 1]
```diff
@@ -158,6 +164,31 @@ function canFillResultSetFromLimitBuffer<RxDocumentType>(s: StateResolveFunction
...
+    return (
+        !isInsert(s) &&
+        isUpdate(s) &&
+        !isDelete(s) &&
+        hasLimit(s) &&
+        !isFindOne(s) &&
+        !hasSkip(s) &&
+        !wasResultsEmpty(s) &&
+        !previousUnknown(s) &&
+        // wasLimitReached(s) && // both of these was LimitReachedCases are bork.
```
Comment (mitchelljustin) [inline src/event-reduce.ts:178]: make this more grammatical please

File: src/event-reduce.ts [hunk 2]
```diff
@@ -158,6 +164,31 @@ function canFillResultSetFromLimitBuffer<RxDocumentType>(s: StateResolveFunction
...
 
+function isBrokenSortedLimitCase<RxDocumentType>(s: StateResolveFunctionInput<RxDocumentType>) {
+    // The issue is specifically with limited, sorted lists having updated items moved to the top.
+    return (
+        !isInsert(s) &&
+        isUpdate(s) &&
+        !isDelete(s) &&
+        hasLimit(s) &&
+        !isFindOne(s) &&
+        !hasSkip(s) &&
```
Comment (mitchelljustin) [inline src/event-reduce.ts:175]: i wonder if this is also happening for queries with a skip in them i.e. query page > 0

File: src/event-reduce.ts [hunk 3]
```diff
@@ -158,6 +164,31 @@ function canFillResultSetFromLimitBuffer<RxDocumentType>(s: StateResolveFunction
     );
 }
 
+function isBrokenSortedLimitCase<RxDocumentType>(s: StateResolveFunctionInput<RxDocumentType>) {
+    // The issue is specifically with limited, sorted lists having updated items moved to the top.
```
Comment (mitchelljustin) [inline src/event-reduce.ts:168]: Link the Linear issue here / Artem's explanation for extra context

**Review by mitchelljustin — APPROVED** [review]
LGTM

File: src/event-reduce.ts (hunk 1 above)
Comment (TristanH) [reply src/event-reduce.ts:178]: no

**Pull request merged**